assets/dev/js/frontend/utils/url-actions.js

```diff
@@ -64,8 +65,13 @@ export default class extends elementorModules.ViewModule {
 		this.runAction( jQuery( event.currentTarget ).attr( 'href' ), event );
 	}
 
+	findElementWithHash() {
+		return document.querySelector( `[e-action-hash="${ location.hash }"]` ) || document.querySelector( `a[href*="${ location.hash }"]` );
+	}
+
 	runHashAction() {
-		if ( location.hash ) {
+		// Look for elements with the action hash attribute.
+		if ( location.hash && this.findElementWithHash() ) {
```

Comment:
Get the hash from the element and use it. Don't use the location.hash

Reply:
Done